src/ol/renderer/canvas/tilelayer.js

```diff
@@ -183,7 +183,8 @@ ol.renderer.canvas.TileLayer.prototype.prepareFrame = function(frameState, layer
       if (this.isDrawableTile_(tile)) {
         if (tile.getState() == ol.TileState.LOADED) {
           tilesToDrawByZ[z][tile.tileCoord.toString()] = tile;
-          if (!newTiles && this.renderedTiles.indexOf(tile) == -1) {
+          var inTransition = tile.getAlpha && tile.getAlpha(ol.getUid(this), frameState.time) !== 1;
+          if (!newTiles && (this.renderedTiles.indexOf(tile) === -1 || inTransition)) {
```

Comment:
The second condition can be flipped to avoid the unnecessary `indexOf` lookup when in transition.